jsonpickle/util.py

```diff
@@ -43,10 +43,19 @@ def is_type(obj):
     >>> is_type(Klass)
     True
     """
+    # use "isinstance" and not "is" to allow for metaclasses
     if PY3:
-        return type(obj) is type
+        return isinstance(obj, type)
     else:
-        return type(obj) is type or type(obj) is types.ClassType
+        return isinstance(obj, (type, types.ClassType))
+
+
+def has_method(obj, name, *args, **kwargs):
+    try:
+        getattr(obj, name)(*args, **kwargs)
+        return True
+    except:
+        return False
```

Comment:
```
$ python
Python 2.7.5 (default, Mar  9 2014, 22:15:05)
[GCC 4.2.1 Compatible Apple LLVM 5.0 (clang-500.0.68)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import six
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named six
```

I know this sounds silly but please avoid the dependency if possible. I know it's a bit of copy/paste boilerplate, but that boilerplate is not changing anytime soon since the 2/3 overlap is a relatively known beast.  compat is there to avoid the dependency and is only 42 lines right now.